src/GameEntityTest.java

```diff
@@ -0,0 +1,193 @@
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+/**
+ * Utility class with implementation of parts of Kernel methods in GmaeEntity.
+ *
+ * @author Joonie Kim
+ *
+ */
+public final class GameEntityTest {
+
+    /**
+     * Private constructor so this utility class cannot be instantiated.
+     */
+    private GameEntityTest() {
+
+    }
+
+    /**
+     * Moves the entity at map[y][x] to forward by its' direction.
+     *
+     * @param map
+     *            map field of the entity exist.
+     * @param x
+     *            x coordinate of the entity.
+     * @param y
+     *            y coordinate of the entity.
+     * @updates map
+     */
+    public static void moveForward(int[][] map, int x, int y) {
+        if (map[y][x] == 1 && map[y].length - 1 > x + 1) {
+            map[y][x + 1] = map[y][x];
+            map[y][x] = 0;
+        } else if (map[y][x] == 2 && map.length - 1 > y + 1) {
+            map[y + 1][x] = map[y][x];
+            map[y][x] = 0;
+        } else if (map[y][x] == 3 && 0 < x) {
+            map[y][x - 1] = map[y][x];
+            map[y][x] = 0;
+        } else if (map[y][x] == 4 && 0 < y) {
+            map[y - 1][x] = map[y][x];
+            map[y][x] = 0;
+        }
+    }
+
+    /**
+     * Turns the entity at map[y][x] to right from its' original direction.
+     *
+     * @param map
+     *            map field of the entity exist.
+     * @param x
+     *            x coordinate of the entity.
+     * @param y
+     *            y coordinate of the entity.
+     * @updates map
+     */
+    public static void turnRight(int[][] map, int x, int y) {
+        map[y][x] += 1;
+        if (map[y][x] > 4) {
+            map[y][x] = 1;
+        }
+    }
+
+    /**
+     * Turns the entity at map[y][x] to left from its' original direction.
+     *
+     * @param map
+     *            map field of the entity exist.
+     * @param x
+     *            x coordinate of the entity.
+     * @param y
+     *            y coordinate of the entity.
+     * @updates map
+     */
+    public static void turnLeft(int[][] map, int x, int y) {
+        map[y][x] -= 1;
+        if (map[y][x] < 0) {
+            map[y][x] = 4;
+        }
+    }
+
+    /**
+     * Turns back the entity at map[y][x] from its' original direction.
+     *
+     * @param map
+     *            map field of the entity exist.
+     * @param x
+     *            x coordinate of the entity.
+     * @param y
+     *            y coordinate of the entity.
+     * @updates map
+     */
+    public static void turnBack(int[][] map, int x, int y) {
```

Comment:
I would make the entity a separate object OR store the x and y globally. These methods probably don't need any parameters if you you're storing the map and position of the entity.